apiserver/facades/controller/firewaller/firewaller.go

```diff
@@ -218,7 +214,15 @@ func (f *FirewallerAPI) ModelFirewallRules(ctx context.Context) (params.IngressR
 	if err != nil {
 		return params.IngressRulesResult{Error: apiservererrors.ServerError(err)}, nil
 	}
-	isController := f.st.IsController()
+
+	controllerModelUUID, err := f.controllerService.ControllerModelUUID(ctx)
```

Comment:
Because this is a model scoped facade we can just ask the current model if it is the controller model. See https://github.com/tlm/juju/blob/f9bb18a7a3e85048c1c7aec60b0312196302643c/core/model/modelinfo.go#L43

This comes off of the model info service.

Reply:
Sure thing, thanks tom!

Reply:
Refactored with thanks!